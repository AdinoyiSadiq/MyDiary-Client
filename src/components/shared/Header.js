import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import HeaderModal from '../HeaderModal';

class Header extends Component {
  state = { showModal: false }

  componentWillMount() {
    this.parseUrl(this.props);
  }

  parseUrl(props) {
    const { match: { path } } = props;
    const route = path.split("main/")[1];
    this.setState({ route });
  }

  highlightIcon(view) {
    const { route } = this.state;
    const current = (route === view ? 'current' : null);
    return current;
  }

  showHeaderModal = () => {
    this.setState(prevState => ({
      showModal: !prevState.showModal
    }));
  }

  render() {
    const { showModal } = this.state;
    return (
      <div>
        <header className="clearfix">
          <nav className="mainNav" role="navigation">
            <ul>
              <li className="logo">
                <h3><Link className="logoLink" to="/main/entries">MyDiary</Link></h3>
              </li>
              <li>
                <Link to="/main/entries" className={`${this.highlightIcon('entries')}`}>Entries</Link>
              </li>
              <li>
                <div 
                  id="profile" 
                  className={`navProfile ${this.highlightIcon('profile')}`}
                  onClick={this.showHeaderModal}
                >
                  Adinoyi
                </div>
              </li>
              <li>
                <Link to="/main/addEntry" className={`${this.highlightIcon('addEntry')}`}>
                  <i className="fas fa-plus" />
                </Link>
              </li>
            </ul>
          </nav>
        </header>
        <HeaderModal showModal={showModal} />
      </div>
    );
  }
};

export default Header;