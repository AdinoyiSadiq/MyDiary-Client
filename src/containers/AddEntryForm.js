import React, { Component } from 'react';
import { connect } from 'react-redux';
import { createEntry } from '../actions/entryActions';
import validateAuth from '../lib/validation';

const initialState = {
  title: '',
  content: '',
  touched: {
    title: false,
    content: false,
  },
};

const fieldNames = ['title', 'content'];

class AddEntryForm extends Component {
  state = initialState;

  handleSubmit = async (event) => {
    event.preventDefault();
    const { createEntry, history } = this.props;
    const { title, content } = this.state;

    const validationError = validateAuth({ title, content }, fieldNames);
    const changedTouchState = this.changeTouchState(fieldNames, true);
    this.setState({ touched: changedTouchState });

    if (!validationError.status) {
      createEntry({ title, content }, (entryId) => history.push(`/main/entry/${entryId}`));
      this.setState(initialState);
    }
  }

  handleChange = (event) => {
    this.setState({ [event.target.id]: event.target.value });
  }
  
  handleBlur = (field) => () => {
    const { touched } = this.state;
    this.setState({
      touched: { ...touched, [field]: true },
    });
  }

  changeTouchState = (fields, state) => {
    const touchState = {};
    fields.forEach((field) => {
      touchState[field] = state;
    });
    return touchState;
  }

  cancelEntry = () => {
    const { history } = this.props;
    this.setState({initialState});
    history.push('/main/entries')
  }

  render() {
    const { title, content, touched } = this.state;
    const { errorMessage } = this.props;
    const error = validateAuth({ title, content }, fieldNames);
    
    return (
      <main>
        <section>
          <form onSubmit={this.handleSubmit}>
          
            <div className="heading">Add an Entry</div>
            <div className="entryError">{errorMessage}</div>
            <input 
              type="text" 
              placeholder="Title" 
              id="title" 
              value={title} 
              onChange={this.handleChange}
              onBlur={this.handleBlur('title')}
            />
            <div className="entryError">{touched.title ? error.title : ''}</div>
            <textarea 
              rows="20" 
              cols="50" 
              placeholder="Content" 
              id="content" 
              value={content} 
              onChange={this.handleChange}
              onBlur={this.handleBlur('content')}
            />
            <div className="entryError">{touched.content ? error.content : ''}</div>
            <button type="submit">Save</button>
            <button type="button" onClick={this.cancelEntry}>Cancel</button>
          </form>
        </section>
      </main>
    )
  }
}

function mapStateToProps(state) {
  return { errorMessage: state.entry.errorMessage };
}

export default connect(mapStateToProps, { createEntry })(AddEntryForm);