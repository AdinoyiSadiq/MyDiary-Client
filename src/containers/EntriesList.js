import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getEntriesList } from '../actions/entriesListActions';
import EmptyListModal from '../components/EmptyListModal';

class EntriesList extends Component {
  componentWillMount() {
    const { getEntriesList } = this.props;
    getEntriesList();
  }

  renderEntriesList(entriesList) {
    const entries = entriesList.map((entry) => {
      const date = new Date(parseInt(entry.created));
      return (
        <article key={entry.id} className="entry">
          <div className="content">
            <Link to={`/main/entry/${entry.id}`}>
              <h3 className="title">{entry.title}</h3>
              <p className="date">{date.toString().substring(0, 15)}</p>
              <p className="entryContent">{(entry.content).substring(0, 100)}</p>
            </Link>
            <div className="actions">
              <div title="Edit"><i className="far fa-edit"></i></div>
              <div title="Delete"><i className="far fa-trash-alt" id="delete"></i></div>
            </div>
          </div>
        </article>
      );
    });
    return (
      <section className="entryList" id="list">
        {entries}
      </section>
    );
  }

  render() {
    const { entries } = this.props;
    if (entries) {
      return (
        <main className="clearfix">
          {(entries.length > 1) ? this.renderEntriesList(entries) : <EmptyListModal />}
        </main>
      );
    }
    return <div>Loading...</div>
  }
};

function mapStateToProps(state) {
  return { entries: state.entriesList.entries };
}

export default connect(mapStateToProps, { getEntriesList })(EntriesList);