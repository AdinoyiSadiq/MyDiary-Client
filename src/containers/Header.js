import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import HeaderModal from '../components/HeaderModal';
import { getUserProfile } from '../actions/profileActions';

export class Header extends Component {
  state = { showModal: false }

  componentWillMount() {
    const { getUserProfile } = this.props;
    getUserProfile();
    this.parseUrl(this.props);
  }

  parseUrl(props) {
    const { match: { path } } = props;
    const route = path.split("main/")[1];
    this.setState({ route });
  }

  highlightIcon(view) {
    const { route } = this.state;
    const current = (route === view ? 'current' : null);
    return current;
  }

  showHeaderModal = () => {
    this.setState(prevState => ({
      showModal: !prevState.showModal
    }));
  }

  render() {
    const { showModal } = this.state;
    const { profile } = this.props;
    return (
      <div>
        <header className="clearfix">
          <nav className="mainNav" role="navigation">
            <ul>
              <li className="logo">
                <h3><Link className="logoLink" to="/main/entries">MyDiary</Link></h3>
              </li>
              <li>
                <Link to="/main/entries" className={`${this.highlightIcon('entries')}`}>Entries</Link>
              </li>
              <li>
                <div 
                  id="profile" 
                  className={`navProfile ${this.highlightIcon('profile')}`}
                  onClick={this.showHeaderModal}
                >
                  {profile ? profile.firstname : 'Profile'}
                </div>
              </li>
              <li>
                <Link to="/main/addEntry" className={`${this.highlightIcon('addEntry')}`}>
                  <i className="fas fa-plus" />
                </Link>
              </li>
            </ul>
          </nav>
        </header>
        <HeaderModal showModal={showModal} />
      </div>
    );
  }
};

function mapStateToProps(state) {
  return { profile: state.userProfile.profile };
}

export default connect(mapStateToProps, { getUserProfile })(Header);