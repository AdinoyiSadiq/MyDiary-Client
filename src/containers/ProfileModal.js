import React, { Component } from 'react';
import { connect } from 'react-redux';
import ProfileToggle from '../components/ProfileToggle';
import ProfileDetails from '../components/ProfileDetails';
import { getUserProfile } from '../actions/profileActions';

class ProfileModal extends Component {
  componentDidMount = () => {
    const { getUserProfile } = this.props;
    getUserProfile();
  }

  renderProfileModal() {
    const { profile } = this.props;
    if (profile) {
      return (
        <section>
          <div>
            <ProfileToggle />
            <ProfileDetails profile={profile} />
          </div>
        </section>
      );
    } else {
      return <div>Loading...</div>
    }
  }

  render() {
    return (
      <main>
        {this.renderProfileModal()}
	    </main>
    );
  }
}

function mapStateToProps(state) {
  return { profile: state.userProfile.profile };
}

export default connect(mapStateToProps, { getUserProfile })(ProfileModal);