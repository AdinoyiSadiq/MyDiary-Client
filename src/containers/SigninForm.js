import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { signin, clearSigninError } from '../actions/authActions';
import validateAuth from '../lib/validation';
import AuthInput from '../components/shared/AuthInput';
import Button from '../components/shared/Button';

const initialState = {
  email: '',
  password: '',
  touched: {
    email: false,
    password: false,
  },
};

const fieldNames = ['email', 'password'];

class SigninForm extends Component {
	state = initialState;

	componentDidMount = () => {
    const { clearSigninError: clearError } = this.props;
    clearError();
  }

	handleSubmit = async (event) => {
    event.preventDefault();
		const { signin: signinUser, history } = this.props;
		const { email, password } = this.state;

		const validationError = validateAuth({ email, password }, fieldNames);
		const changedTouchState = this.changeTouchState(fieldNames, true);
		this.setState({ touched: changedTouchState });

		if (!validationError.status) {
      signinUser({ email, password }, () => history.push('/main/entries'));
      this.setState(initialState);
    }
  }

	handleChange = (event) => {
    this.setState({ [event.target.id]: event.target.value });
	}
	
	handleBlur = (field) => () => {
    const { touched } = this.state;
    this.setState({
      touched: { ...touched, [field]: true },
    });
  }

  changeTouchState = (fields, state) => {
    const touchState = {};
    fields.forEach((field) => {
      touchState[field] = state;
    });
    return touchState;
  }
	
	render() {
		const { email, password, touched } = this.state;
		const { errorMessage } = this.props;
		const validationError = validateAuth({ email, password }, fieldNames);

		return (
			<div>
				<div className="authContent signin">
					<main>
						<div className="authLogo">MyDiary</div>
						<div className="title">Welcome back!</div>
						<hr/>
						<form className="signinForm" onSubmit={this.handleSubmit}>
							<div className="authError">{errorMessage}</div>
							<AuthInput
								value={email}
								touched={touched}
								error={validationError}
								handleChange={this.handleChange} 
								handleBlur={this.handleBlur('email')}
								name="email" 
								placeholder="Email" 
							/>
							<AuthInput
								value={password}
								touched={touched}
								error={validationError}
								handleChange={this.handleChange}
								handleBlur={this.handleBlur('password')} 
								name="password"
								type="password" 
								placeholder="Password" 
							/>
							<Button title="Sign In" type="submit" />
						</form>
					</main>
				</div>
				<div className="authAlt">
					Don't have an account?
					<Link to="/auth/signup">Sign Up</Link>
				</div>
			</div>
		);
	}
};

function mapStateToProps(state) {
  return { errorMessage: state.auth.errorMessage };
}

export default connect(mapStateToProps, { signin, clearSigninError })(SigninForm);