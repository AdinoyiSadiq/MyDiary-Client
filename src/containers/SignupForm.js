import React, {Component} from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { signup } from '../actions/authActions';
import validateAuth from '../lib/validation';
import AuthInput from '../components/shared/AuthInput';
import Button from '../components/shared/Button';

const initialState = {
  firstname: '',
  lastname: '',
  email: '',
  password: '',
  confirmPassword: '',
  touched: {
    firstname: false,
    lastname: false,
    email: false,
    password: false,
    confirmPassword: false,
  },
};

const fieldNames = ['firstname', 'lastname', 'email', 'password', 'confirmPassword'];

class SignupForm extends Component {
	state = initialState;

	handleSubmit = async (event) => {
    event.preventDefault();
		const { signup: signupUser, history } = this.props;
		const {
      firstname, lastname, email, password, confirmPassword,
		} = this.state;
		
		const fields = {
      firstname, lastname, email, password, confirmPassword,
    };
		
		const validationError = validateAuth(fields, fieldNames);
    const changedTouchState = this.changeTouchState(fieldNames, true);
    this.setState({ touched: changedTouchState });

		const body = {
      firstName: firstname.trim(),
      lastName: lastname.trim(),
      email: email.trim(),
      password: password.trim(),
		};
		
		if (!validationError.status) {
      signupUser(body, () => history.push('/main/entries'));
      this.setState(initialState);
    }
  }

	handleChange = (event) => {
    this.setState({ [event.target.id]: event.target.value });
	}
	
	handleBlur = (field) => () => {
    const { touched } = this.state;
    this.setState({
      touched: { ...touched, [field]: true },
    });
	}
	
	changeTouchState = (fields, state) => {
    const touchState = {};
    fields.forEach((field) => {
      touchState[field] = state;
    });
    return touchState;
  }

	render() {
		const {
      firstname, lastname, email, password, confirmPassword, touched, 
		} = this.state;
		const validationError = validateAuth({
      firstname, lastname, email, password, confirmPassword,
    }, fieldNames);

		return (
			<div>
				<div className="authContent signup">
					<main>
						<div className="authLogo">MyDiary</div>
						<div className="title">Create your MyDiary account</div>
						<hr/>
						<form className="signupForm" onSubmit={this.handleSubmit}>
							<AuthInput
								value={firstname}
								touched={touched}
								error={validationError}
								handleChange={this.handleChange}
								handleBlur={this.handleBlur('firstname')} 
								name="firstname" 
								placeholder="First name" 
							/>
							<AuthInput
								value={lastname}
								touched={touched}
								error={validationError}
								handleChange={this.handleChange} 
								handleBlur={this.handleBlur('lastname')} 
								name="lastname" 
								placeholder="Last name" 
							/>
							<AuthInput
								value={email}
								touched={touched}
								error={validationError}
								handleChange={this.handleChange} 
								handleBlur={this.handleBlur('email')} 
								name="email" 
								placeholder="Email" 
							/>
							<AuthInput
								value={password}
								touched={touched}
								error={validationError}
								handleChange={this.handleChange} 
								handleBlur={this.handleBlur('password')} 
								name="password"
								type="password" 
								placeholder="Password" 
							/>
							<AuthInput
								value={confirmPassword}
								touched={touched}
								error={validationError}
								handleChange={this.handleChange} 
								handleBlur={this.handleBlur('confirmPassword')} 
								name="confirmPassword" 
								type="password"
								placeholder="Confirm Password"
							/>
							<Button title="Sign Up" type="submit" />
						</form>
					</main>
				</div>
				<div className="authAlt">
					Already have an account?
					<Link to="/auth/signin">Sign In</Link>
				</div>
			</div>
		);
	}
};

export default connect(null, { signup })(SignupForm);