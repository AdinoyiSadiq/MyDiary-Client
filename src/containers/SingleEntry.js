import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getSingleEntry } from '../actions/entryActions';

class SingleEntry extends Component {
  componentWillMount() {
    const { getSingleEntry, match: { params: { id } } } = this.props;
    getSingleEntry(id);
  }

  renderSingleEntry() {
    const { entry } = this.props;
    if (entry) {
      const date = new Date(parseInt(entry.created));
      return (
        <article className="entry">
          <h3 className="title">{entry.title}</h3>
          <p className="date">{date.toString().substring(0, 15)}</p>
          <p className="content">{entry.content}</p>
          <div className="actions">
            <div title="Edit">
              <Link to={`/main/updateEntry/${entry.id}`}>
                <i className="far fa-edit"></i>
              </Link>
            </div>
					  <div title="Delete"><i className="far fa-trash-alt"></i></div>
          </div>
        </article>
      );
    }
    return <div>Loading...</div>
  }

  render() {
    return (
      <main>
        {this.renderSingleEntry()}
      </main>
    );
  }
};

function mapStateToProps(state) {
  return { entry: state.singleEntry.entry };
}

export default connect(mapStateToProps, { getSingleEntry })(SingleEntry);