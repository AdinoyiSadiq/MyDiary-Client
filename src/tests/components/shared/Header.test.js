import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Header } from '../../../containers/Header';

let wrapped;

const match = {
  path: '/main/signin'
}

const profile = {
  firstname: 'firstname'
}

const getUserProfile = jest.fn();

beforeEach(() => {
  wrapped = shallow(<Header match={match} profile={profile} getUserProfile={getUserProfile} />);
});


describe('Header UI', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});