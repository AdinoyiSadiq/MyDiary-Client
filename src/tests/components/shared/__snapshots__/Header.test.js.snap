// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header UI render features view should render as expected 1`] = `
<div>
  <header
    className="clearfix"
  >
    <nav
      className="mainNav"
      role="navigation"
    >
      <ul>
        <li
          className="logo"
        >
          <h3>
            <Link
              className="logoLink"
              replace={false}
              to="/main/entries"
            >
              MyDiary
            </Link>
          </h3>
        </li>
        <li>
          <Link
            className="null"
            replace={false}
            to="/main/entries"
          >
            Entries
          </Link>
        </li>
        <li>
          <div
            className="navProfile null"
            id="profile"
            onClick={[Function]}
          >
            firstname
          </div>
        </li>
        <li>
          <Link
            className="null"
            replace={false}
            to="/main/addEntry"
          >
            <i
              className="fas fa-plus"
            />
          </Link>
        </li>
      </ul>
    </nav>
  </header>
  <Connect(HeaderModal)
    showModal={false}
  />
</div>
`;
